apps/go-mod-validator/src/github.ts

```diff
@@ -64,19 +61,39 @@ async function isTagInDefaultBranch(
       commitSha,
     });
   } catch (e) {
-    core.error(`Error checking tag in default branch: ${e}`);
+    let eStr = "unknown";
+    if (e instanceof Error) {
+      eStr = e.message;
+    } else if (typeof e === "string") {
+      eStr = e;
+    }
+
     return {
-      isInDefault: false,
+      isInDefault: "unknown",
       commitSha,
+      reason: eStr,
     };
   }
 }
 
-interface GoModDefaultBranchLookupResult {
+interface GoModDefaultBranchKnownLookupResult {
   isInDefault: boolean;
   commitSha: string;
 }
 
+interface GoModDefaultBranchUnknownResult {
+  isInDefault: "unknown";
+  /**
+   * Additional information when "isInDefault" is "unknown"
+   */
+  reason: string;
+  commitSha: string;
+}
+
+export type GoModDefaultBranchLookupResult =
+  | GoModDefaultBranchKnownLookupResult
+  | GoModDefaultBranchUnknownResult;
```

Comment:
Would an enum for `isInDefault` be better than this union type?

Reply:
With an enum, we'll still have a union type if we want to perform a discriminated union check to differentiate between an `unknown` state vs a `known` one. I initially was going to use an enum but TBH its overkill ATM.